contracts/governance/utils/Votes.sol

```diff
@@ -130,14 +143,19 @@ abstract contract Votes is Context, EIP712, Nonces, IERC5805 {
         bytes32 r,
         bytes32 s
     ) public virtual override {
-        require(block.timestamp <= expiry, "Votes: signature expired");
+        if (block.timestamp > expiry) {
+            revert VotesExpiredSignature(expiry);
```

Comment:
I'm still analyzing them, this is a good candidate so I'll keep note.

My main problem with the errors is that some of them look generic but may require different parameters depending on the context, for example, here the expiry is a `block.timestamp` but it can easily be `block.number`, that info is lost by generalizing the error.

Maybe I'm just worrying too much, what do you think?

Reply:
Suppose we create a common `ExpiredSignature` error and later we want a similar error for block number, I would just use a different name for the second one we add.